src/validation/rules/NoUnusedVariablesRule.ts

```diff
@@ -17,7 +17,10 @@ export function NoUnusedVariablesRule(context: ValidationContext): ASTVisitor {
     OperationDefinition(operation) {
       const usages = context.getRecursiveVariableUsages(operation);
       const variableNameUsed = new Set<string>(
-        usages.map(({ node }) => node.name.value),
+        usages
+          // Skip variables used as fragment arguments
+          .filter(({ fragmentVarDef }) => !fragmentVarDef)
+          .map(({ node }) => node.name.value),
       );
```

Comment:
I noticed these get skipped over in NoUndefinedFragmentVariableRule, maybe the better place to edit is changing `getRecursiveVariableUsages()` to something like `getOperationVariableUsages()`?

Reply:
If we're OK making a breaking change. But Meta has custom validation rules that use `getRecursiveVariableUsages` so I imagine other people likely do too?

Reply:
Or leave the existing method behind (optionally deprecate it if its not used in core anymore?) and introduce this new one?